app/main/views/services.py

```diff
@@ -102,24 +101,36 @@ def update_service_status(service_id):
 @login_required
 @flask_featureflags.is_active_feature('EDIT_SERVICE_PAGE')
 def edit_section(service_id, section):
-
-    service = data_api_client.get_service(service_id)['services']
+    service = data_api_client.get_service(service_id)
+    if service is None:
+        abort(404)
+    service = service['services']
 
     if not _is_service_associated_with_supplier(service):
         abort(404)
 
     content = existing_service_content.get_builder().filter(service)
+    section = content.get_section(section)
+    if section is None:
+        abort(404)
 
     return render_template(
         "services/edit_section.html",
-        section=content.get_section(section),
+        section=section,
+        page_questions='|'.join(get_section_questions(section)),
```

Comment:
That is a very good point. I have updated the PR to not pass the `page_questions` to the template and only use the ones from the section when needed.